src/test/ui/lint/empty-iterator-range.stderr

```diff
@@ -0,0 +1,26 @@
+error: this `for` loop is never run
+  --> $DIR/empty-iterator-range.rs:4:5
+   |
+LL |     for _i in 10..0 {
+   |     ^^^^^^^^^^^^^^^
```

Comment:
I would love it if you could add a primary label and a note explaining why

```suggestion
error: this `for` loop is never run
  --> $DIR/empty-iterator-range.rs:4:5
   |
LL |     for _i in 10..0 {
   |     ^^^^^^^^^^^^^^^ this loop is never run because it's iterating over an empty iterator
   = note: ranges that have a bigger start than their end will produce an empty iterator
   = help: if you want a decreasing range sequence, create an increasing range and call `.rev()` on it
```

Ideally we'd give a structured suggestion. At this point you no longer have access to the range literal, right? If we did we could have

```
error: this `for` loop is never run
  --> $DIR/empty-iterator-range.rs:4:5
   |
LL |     for _i in 10..0 {
   |     ^^^^^^^^^^^^^^^ this loop is never run because it's iterating over an empty iterator
   = note: ranges that have a bigger start than their end will produce an empty iterator
help: if you want a decreasing range sequence, create an increasing range and call `.rev()` on it
   |
LL |     for _i in (0..10).rev() {
   |               +~  ~~+++++++
```